apps/emqx_management/src/emqx_mgmt_api_subscriptions.erl

```diff
@@ -421,6 +431,31 @@ merge_queries(QString0, Q1, Q2) ->
             }
     end.
 
+maybe_fetch_from_second_query(Params) ->
+    #{
+        rows1 := Data1,
+        limit := Limit,
+        hasnext1 := HN1,
+        meta := Meta,
+        count2 := C2,
+        query2 := Q2,
+        query_string := QString0
+    } = Params,
+    NumRows1 = length(Data1),
+    {Data, HN} =
+        case NumRows1 >= Limit of
+            true ->
+                {Data1, HN1 orelse C2 > 0};
+            false ->
+                #{data := Data2, meta := #{hasnext := HN2}} =
+                    Q2(QString0#{<<"limit">> := Limit - NumRows1}),
+                {Data1 ++ Data2, HN1 or HN2}
```

Comment:
Curious: if we assume that HN1 may be true, why do we use the second query?

Reply:
In this arm of the `case`, the number of returned rows from `Q1` is fewer than `Limit`, so I think we might assume `HN1 = false` in most cases.  But I think you have a good point: this situation is not necessarily wrong, I'll check for this condition.  Thanks!  :beers:

Reply:
:heavy_check_mark: